engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -181,119 +187,143 @@ private void updateTick(float deltaInSeconds) {
+            renderChunksAlphaReject();
+            renderOverlays();
+            renderFirstPersonView();
 
-        for (RenderSystem renderer : systemManager.iterateRenderSubscribers()) {
-            renderer.renderAlphaBlend();
+            DefaultRenderingProcess.getInstance().endRenderSceneOpaque();
+
+            renderLightGeometry();
+            renderChunksRefractiveReflective();
         }
 
-        glDisable(GL_BLEND);
-        glDepthMask(true);
+        postRenderCleanup();
 
-        DefaultRenderingProcess.getInstance().endRenderSceneOpaque();
-        PerformanceMonitor.endActivity();
+        combineRefractiveReflectiveAndOpaquePasses();
+        renderSimpleBlendMaterialsIntoCombinedPass();
 
-        /* RENDER THE FINAL POST-PROCESSED SCENE */
-        PerformanceMonitor.startActivity("Render Post-Processing");
-        DefaultRenderingProcess.getInstance().renderPost(stereoRenderState);
-        PerformanceMonitor.endActivity();
+        renderFinalPostProcessedScene(stereoRenderState);
 
         playerCamera.updatePrevViewProjectionMatrix();
-
     }
 
     private void renderShadowMap() {
-        PerformanceMonitor.startActivity("Render World (Shadow Map)");
+        if (renderingConfig.isDynamicShadows() && isFirstRenderingStageForCurrentFrame) {
+            PerformanceMonitor.startActivity("Render World (Shadow Map)");
 
-        glDisable(GL_CULL_FACE);
+            DefaultRenderingProcess.getInstance().beginRenderSceneShadowMap();
+            glDisable(GL_CULL_FACE);
 
-        shadowMapCamera.lookThrough();
+            shadowMapCamera.lookThrough();
 
-        while (renderQueues.chunksOpaqueShadow.size() > 0) {
-            renderChunk(renderQueues.chunksOpaqueShadow.poll(), ChunkMesh.RenderPhase.OPAQUE, shadowMapCamera, ChunkRenderMode.SHADOW_MAP);
-        }
+            while (renderQueues.chunksOpaqueShadow.size() > 0) {
+                renderChunk(renderQueues.chunksOpaqueShadow.poll(), ChunkMesh.RenderPhase.OPAQUE, shadowMapCamera, ChunkRenderMode.SHADOW_MAP);
+            }
 
-        for (RenderSystem renderer : systemManager.iterateRenderSubscribers()) {
-            renderer.renderShadows();
-        }
+            for (RenderSystem renderer : systemManager.iterateRenderSubscribers()) {
+                renderer.renderShadows();
+            }
 
-        glEnable(GL_CULL_FACE);
+            glEnable(GL_CULL_FACE);
+            DefaultRenderingProcess.getInstance().endRenderSceneShadowMap();
 
-        PerformanceMonitor.endActivity();
+            PerformanceMonitor.endActivity();
+        }
     }
 
     public void renderWorldReflection() {
         PerformanceMonitor.startActivity("Render World (Reflection)");
+
+        DefaultRenderingProcess.getInstance().beginRenderReflectedScene();
+        glCullFace(GL11.GL_FRONT);
+        playerCamera.setReflected(true);
+
         playerCamera.lookThroughNormalized();
         skysphere.render(playerCamera);
+        playerCamera.lookThrough();
```

Comment:
Moved this lookThrough() statement here, as by my understanding this is really only something that needs to be done and then undone for the skysphere. 

Theoretically the previous position should have generated a problem when the water wasn't reflective and the lookThrough() line didn't get called. In practice the next thing being rendered is the skysphere proper, which calls again both lookThroughNormalized() and lookThrough().

Also theoretically this is a performance hit: originally the code had three lookThrough*() calls instead of four. I think at this stage the loss in performance remains unnoticeable and the gain in readability is preferable.